File: app/src/main/java/com/grassehh/app/configuration/ServerConfiguration.java
package com.grassehh.app.configuration;

import io.micrometer.context.ContextSnapshotFactory;
import org.springframework.boot.web.embedded.netty.NettyServerCustomizer;
import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.zalando.logbook.Logbook;
import org.zalando.logbook.netty.LogbookServerHandler;

@Configuration
public class ServerConfiguration {
    @Bean
    public NettyServerCustomizer logbookNettyServerCustomizer(Logbook logbook, ContextSnapshotFactory contextSnapshotFactory) {
        return server -> server.doOnConnection(connection ->
                connection.addHandlerLast(
                        new TracingChannelDuplexHandler(
                                new LogbookServerHandler(logbook),
                                contextSnapshotFactory
                        )
                )
        ).metrics(true, (uriTagValue -> uriTagValue));
    }
}

File: app/src/main/kotlin/com/grassehh/app/configuration/ServerConfiguration.kt
package com.grassehh.app.configuration

import io.micrometer.context.ContextSnapshotFactory
import org.springframework.boot.web.embedded.netty.NettyServerCustomizer
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.zalando.logbook.Logbook
import org.zalando.logbook.netty.LogbookServerHandler
import reactor.netty.Connection

@Configuration
class ServerConfiguration {

    @Bean
    fun logbookNettyServerCustomizer(logbook: Logbook, contextSnapshotFactory: ContextSnapshotFactory) =
        NettyServerCustomizer { server ->
            server.doOnConnection { connection: Connection ->
                connection.addHandlerLast(
                    TracingChannelDuplexHandler(
                        LogbookServerHandler(logbook),
                        contextSnapshotFactory
                    )
                )
            }.metrics(true) { uriTagValue: String ->
                uriTagValue
            }
        }
}